Core/Graphics/Source/FrameBuffer.cpp

```diff
@@ -134,7 +135,7 @@ namespace Babylon::Graphics
             return;
         }
 
-        if (bgfx::isValid(m_handle))
+        if (bgfx::isValid(m_handle) && m_graphicsID == Babylon::Graphics::Device::GetID())
```

Comment:
nit: Can be simplified since we are already in the namespace? Ditto, elsewhere.

```suggestion
        if (bgfx::isValid(m_handle) && m_graphicsID == Device::GetID())
```